src/super_gradients/training/models/detection_models/yolov5_base.py

```diff
@@ -243,35 +248,39 @@ def __init__(self, arch_params):
                                                                                     arch_params.width_mult_factor,
                                                                                     arch_params.depth_mult_factor)
 
+        DownConv = DepthWiseConv if depthwise else Conv
+
         self._modules_list = nn.ModuleList()
         self._modules_list.append(Conv(width_mult(connector[0]), width_mult(512), 1, 1, activation_type))  # 10
         self._modules_list.append(nn.Upsample(None, 2, 'nearest'))  # 11
         self._modules_list.append(Concat(1))  # 12
         self._modules_list.append(
-            block(width_mult(connector[1]), width_mult(512), depth_mult(3), activation_type, False))  # 13
+            block(width_mult(connector[1]), width_mult(512), depth_mult(3), activation_type, False, depthwise))  # 13
 
         self._modules_list.append(Conv(width_mult(512), width_mult(256), 1, 1, activation_type))  # 14
         self._modules_list.append(nn.Upsample(None, 2, 'nearest'))  # 15
         self._modules_list.append(Concat(1))  # 16
         self._modules_list.append(
-            block(width_mult(connector[2]), width_mult(256), depth_mult(3), activation_type, False))  # 17
+            block(width_mult(connector[2]), width_mult(256), depth_mult(3), activation_type, False, depthwise))  # 17
 
-        self._modules_list.append(Conv(width_mult(256), width_mult(256), 3, 2, activation_type))  # 18
+        self._modules_list.append(DownConv(width_mult(256), width_mult(256), 3, 2, activation_type))  # 18
         self._modules_list.append(Concat(1))  # 19
-        self._modules_list.append(block(width_mult(512), width_mult(512), depth_mult(3), activation_type, False))  # 20
+        self._modules_list.append(
+            block(width_mult(512), width_mult(512), depth_mult(3), activation_type, False, depthwise))  # 20
 
-        self._modules_list.append(Conv(width_mult(512), width_mult(512), 3, 2, activation_type))  # 21
+        self._modules_list.append(DownConv(width_mult(512), width_mult(512), 3, 2, activation_type))  # 21
         self._modules_list.append(Concat(1))  # 22
         self._modules_list.append(
-            block(width_mult(1024), width_mult(1024), depth_mult(3), activation_type, False))  # 23
+            block(width_mult(1024), width_mult(1024), depth_mult(3), activation_type, False, depthwise))  # 23
 
         if arch_params.yolo_type != 'yoloX':
             self._modules_list.append(
                 Detect(num_classes, anchors, channels=[width_mult(v) for v in (256, 512, 1024)]))  # 24
         else:
             strides = anchors.stride
             self._modules_list.append(
-                DetectX(num_classes, strides, activation_type, channels=[width_mult(v) for v in (256, 512, 1024)]))  # 24
+                DetectX(num_classes, strides, activation_type, channels=[width_mult(v) for v in (256, 512, 1024)],
+                        depthwise=arch_params.depthwise))  # 24
```

Comment:
no need to check `arch_params.depthwise`. you already have local variable `depthwise` here